specification/loadtestservice/data-plane/Microsoft.LoadTestService/stable/2022-11-01/loadtestservice.json

```diff
@@ -0,0 +1,2962 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Load Testing",
+    "description": "These APIs allow end users to create, view and run load tests using Azure Load Test Service.",
+    "version": "2022-11-01"
+  },
+  "paths": {
+    "/appcomponents/{name}": {
+      "patch": {
```

Comment:
@mikekistler I am adding more context to this operation. We store customer azure resource details as configuration so that we can pull azure resource metrics, along with our load test metrics and this helps customers to compare the load test effect on their application resources. AppComponent entity stores a list of azure resource details (like resource-id, type, name, etc.). This is required to pull the metrics info from the azure monitor. This app component can be attached with a test or test run. So for this operation, we take AppComponent ( which is basically a map of azure resources) & test, test run id to associate with, and combine all this we call as AppcomponentMap. 
Considering this flow, let us know if the request body and response are correct or not. Let us know if there is a better way to design this API as well.

Reply:
Let me explain my comment a bit more. The convention for REST APIs is that when a path ends in a static path segment, like "app-components", that last segment is the name of a "collection" of resources. It is always plural, because it is a collection, but the resources of the collection are individual instances of that resource, so in this case "app-component".

It seems that you are trying to use this endpoint to define a "collection of collections" -- this is NOT conventional in REST.

And it is even more confusing because the description of the operation says it associates an "app component (Azure resource)" (singular) "to a test or test run".

One possible solution is to simply embed the app components map directly into the Test and TestRuns schemas. Just make it a property of these other resource models. This would simplify your API by eliminating all the /app-component operations. 

Another possibility is to make "app-components" a sub-path of /loadtests/{testId} and also /testruns/{testRunId} with get and patch methods (and possibly a delete method -- not sure that's needed).

Reply:
We have implemented the suggested approach